client/components/data/query-theme-details/index.js

```diff
@@ -0,0 +1,55 @@
+/** @ssr-ready **/
+
+/**
+ * External dependencies
+ */
+import { PropTypes, Component } from 'react';
+import { connect } from 'react-redux';
+
+/**
+ * Internal dependencies
+ */
+import { isRequestingThemeDetails } from 'state/themes/theme-details/selectors';
+import { fetchThemeDetails } from 'state/themes/actions';
+
+/**
+ * Fetches details for a theme specified by its ID
+ * and passes it to the supplied child component.
+ */
+class QueryThemeDetails extends Component {
+
+	componentDidMount() {
+		this.refresh( this.props );
+	}
+
+	componentWillReceiveProps( nextProps ) {
+		if ( nextProps.id && nextProps.id !== this.props.id ) {
+			this.refresh( nextProps );
+		}
+	}
+
+	refresh( props ) {
+		// todo (seear): Don't fetch if site matches existing data
```

Comment:
I think we can remove this todo. It was put in because we used to have an optimization here. We haven't missed the optimization, so we may as well keep it simple.